hoodie-client/src/main/java/com/uber/hoodie/io/HoodieAppendHandle.java

```diff
@@ -128,8 +134,9 @@ private void init(Iterator<HoodieRecord<T>> recordItr) {
       }
       // update the new location of the record, so we know where to find it next
       record.setNewLocation(new HoodieRecordLocation(commitTime, fileId));
-    });
-    this.records = records;
+      averageRecordSize = SizeEstimator.estimate(record);
+      addAndCheckIfFlushNeeded(record, (int) (config.getLogFileDataBlockMaxSize() / averageRecordSize));
```

Comment:
I feel it will be cleaner to move this into `doAppend()` and do the initial samples there based on a local boolean flag.. The addtional call to `addAndCheckIfFlushNeeded` is a little harder to grok

Reply:
I'm fine with moving this to append.

Reply:
Looks like we are prefer different things for the same reason :).  From where I am coming from, it would be good to see in a place what state is kept.  (like modularizing of the code into the addAndCheck... method)